experimental/sql/dialects/rel_alg.py

```diff
@@ -112,6 +112,23 @@ class Nullable(DataType):
   type: ParameterDef[DataType]
 
 
+@irdl_attr_definition
+class Order(ParametrizedAttribute):
```

Comment:
The result "type" of `Order` is a big different from the other operators in that it returns a sequence, not a bag. There *are* probably ways to not represent that with a type but we have to think a bit about the implications and semantics of having this op.

Reply:
I feel that this is not important atm, since we do not have types in this dialect. But I agree that I have to think of something for the ssa-based dialects.

Reply:
Yes, you are right. This is (probably) another downside of this style of tree-shaped dialect and representing information as implicitly as we are atm. In the current design, it might just end up being another tree traversal, trying to figure out whether the stream is order at a given operation.

I have however defined an optimized compilation stack and there we can have this information on the tree-level. Also, this actually gives me a great example for why this representing types on the tree level is useful, thanks a lot! :)

(Spoiler: in the new design, the tree comes in a type-agnostic and a type-aware version, so SSAifying and type inference are split up. This means that, e.g., projection pushwodn can be down type-agnosticly, whereas optimizations that rely on the order can be done on the type-aware tree dialect).